File: musicShopProject/Model/Users/UserBlank.cs
namespace musicShopProject.Model.Users;

public partial class UserBlank
{
    public Guid? Id { get; set; }
    public String? PhoneNumber { get; set; }
    public String? Password { get; set; }
    public Boolean PasswordBeChanged { get; set; }
}

public partial class UserBlank
{
    public class Validated
    {
        public Guid Id { get; }
        public String PhoneNumber { get; }
        public Password Password { get; }
        public Boolean PasswordBeChanged { get; }

        public Validated(Guid id, Password password, String phoneNumber, Boolean passwordBeChanged)
        {
            Id = id;
            Password = password;
            PhoneNumber = phoneNumber;
            PasswordBeChanged = passwordBeChanged;
        }
    }
}

File: musicShopProject/tentashop/src/domain/user/userBlank.ts
import { User } from "./user";

export interface UserBlank{
    id: string | null,
    phoneNumber: string | null,
    password: string | null,
    PasswordBeChanged: boolean | null
}

export namespace UserBlank {
    export function getEmpty(): UserBlank {
        return {
            id: null,
            phoneNumber: null,
            password: null,
            PasswordBeChanged: null
        };
    }

    export function toUserBLank(user: User): UserBlank {
        return {
            id: user.id,
            phoneNumber: user.phoneNumber,
            password: null,
            PasswordBeChanged: false
        };
    }
 }
